packages/transaction-controller/src/utils/history.ts

```diff
@@ -40,16 +56,99 @@ export function updateTransactionHistory(
 
   const currentState = snapshotFromTransactionMeta(transactionMeta);
   const previousState = replayHistory(transactionMeta.history);
-  const historyEntry = generateHistoryEntry(previousState, currentState, note);
+  const newHistoryEntry = generateHistoryEntry(
+    previousState,
+    currentState,
+    note,
+  );
+
+  if (newHistoryEntry.length > 0) {
```

Comment:
Minor, could we early return here instead so the core logic isn't nested in a block?

Reply:
Good idea, done in fffcd212

I had missed this because this was pre-existing code, the diff makes it seem new only because I renamed `newHistoryEntry`. But that's fine, this is nicer.